src/hasher512.cpp

```diff
@@ -195,10 +183,11 @@ namespace libtorrent
 		}
 		TORRENT_ASSERT(size == digest.size());
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Final(reinterpret_cast<unsigned char*>(digest.begin()), &m_context);
+		SHA512_Final(digest.begin(), &m_context);
```

Comment:
I think `data()` is a bit better than `begin()`, since in principle, `begin()` could return an iterator